### ISSUE TITLE
Error: can't find module main

### ISSUE THREAD

**Issue by jmstallard** [awslabs/lambda-chef-node-cleanup#13]
Hi AWSLabs:
I followed the instructions using terraform to upload the payload. Everything look like it worked (assignment of site-specific Environment variables, region, etc) and included encrypted chef key.

In the console, I see all the attributes as expected, including main.handle as handler - all done by terraform.
I do have VPC configured as well & applied the appropriate policy per instructions.

However, when running the test, I get:

START RequestId: 9160bb9a-1a24-11e7-8b2c-adbf48dd1b21 Version: $LATEST
Unable to import module 'main': No module named main

Any ideas?

**Comment by z-matth** [comment]
+1 
same thing for me. played around with all kinds of module/handler names. nothing worked.
Unable to import module 'main': No module named chef

**Comment by z-matth** [comment]
@jmstallard - I fixed this for me by making sure i was zipping properly. this instructions for this are not the most robust, so I feel your pain. when you zip the lambda function make sure to pass the "-r" flag for recursive. i mistakenly thought zip would be recursive by default. which (for me) was not.

For ex:
```bash
#!/bin/bash
PROFILE=my-aws-profile
NAME=my-awesome-function

zip -r $NAME.zip ./*
aws lambda create-function --cli-input-json file://$NAME.json --zip-file fileb://$NAME.zip --profile $PROFILE && rm $NAME.zip
```

Where the json file looks something like this:
```json
{
    "FunctionName": "<my-awesome-function>",
    "Runtime": "python2.7",
    "Role": "arn:aws:iam::############:role/<my-role-name>",
    "Handler": "main.handle",
    "Description": "<your description here>",
    "Timeout": 4,
    "MemorySize": 128,
    "Publish": true
}
```

**Comment by jmstallard** [comment]
Hey, Thanks for the follow-up, z-matt...

I was using the terraform & did zip up as per doc:

zip -r lambda_function_payload.zip lambda/*


Just tried this and function is now found:

cd lambda

zip -r ../lambda_function_payload.zip *

Function is now found! ... but doesn't work. times out without error after 90sec.
Oh well, a higher-class of error I guess.

-JMS.

________________________________

**Comment by ranman** [comment]
@jmstallard try the ranman-patch-1 branch -- there was an error in the code (see issue #16)

**Comment by seajoshc** [comment]
Fix for #16 has been merged in, thanks @ranman.
